app/controllers/assignment_invitations_controller.rb

```diff
@@ -35,6 +48,24 @@ def check_user_not_previous_acceptee
     redirect_to success_assignment_invitation_path
   end
 
+  def check_authorized_repo_setup
+    redirect_to success_assignment_invitation_path unless repo_setup_enabled?
+  end
+
+  def classroom_config
+    starter_code_repo_id = current_submission.assignment.starter_code_repo_id
+    client               = current_submission.creator.github_client
+
+    starter_repo         = GitHubRepository.new(client, starter_code_repo_id)
+    ClassroomConfig.new(starter_repo)
+  end
+
+  def configurable_submission?
+    repo = current_submission.github_repository
+    configurable = classroom_config.configurable? repo
+    repo.import_progress[:status] == "complete" && configurable && params[:configure]
```

Comment:
What does params[:configure] here represent? It might be nice to wrap this in a method to describe what it represents

```
def something_meaningful
  params[:configure]
end
```